### ISSUE TITLE
Implement State Based Router

### ISSUE THREAD

**Issue by sfount** [third-culture-software/bhima#206]
The need for correctly handling state in application modules has been recorded for a while - https://github.com/IMA-WorldHealth/bhima-2.X/issues/51. 

At this point many configuration (management modules) have been developed and is essential that a good standard for how to proceed with module routing is developed. There are many reviews on code that notice a weakness in state management. 

Routing should either be implemented with UI-Router or Component Router (towards Angular 2); with a proof of concept implemented as soon as possible. 

Suggested course of action: 
1. Module is selected for proof of concept implementation
2. Module is developed using the new utility (UI Router/ Component Router) 
3. Code is reviewed 
4. Once the code is merged into master a wiki article defines exactly how to handle state in an application module and this is then used to measure future code merge requests. 
@IMA-WorldHealth/local-contributors

**Comment by jniles** [comment]
I'm on it.
